modular_skyrat/modules/time_clock/code/console.dm

```diff
@@ -0,0 +1,248 @@
+		return TRUE
+
+	return FALSE
+
+///Is the inserted ID on cooldown? returns TRUE if the ID has a cooldown
+/obj/machinery/time_clock/proc/id_cooldown_check()
+	if(!inserted_id)
+		return FALSE
+
+	var/datum/component/off_duty_timer/id_component = inserted_id.GetComponent(/datum/component/off_duty_timer)
+	if(id_component && id_component.hop_locked)
+		return TRUE
+
+	if(!id_component || !id_component.on_cooldown)
+		return FALSE
+
+	return TRUE
+
+///Is the inserted ID off-duty? Returns true if the ID is off-duty
+/obj/machinery/time_clock/proc/off_duty_check()
+	if(!inserted_id)
+		return FALSE
+
+	var/datum/component/off_duty_timer/id_component = inserted_id.GetComponent(/datum/component/off_duty_timer)
+	if(!id_component)
+		return FALSE
+
+	return TRUE
+
+/datum/component/off_duty_timer
+	///Is the ID that the component is attached to is able to clock back in?
+	var/on_cooldown = FALSE
+	///The stored ID trim of the user of the id
+	var/datum/id_trim/job/stored_trim
+	///Is the owner of card locked out of clocking back in until their ID is unlocked by the HoP?
+	var/hop_locked = FALSE
+	///What was the name of the job the person was working when they clocked out?
+	var/stored_assignment = ""
+
+/datum/component/off_duty_timer/Initialize(cooldown_timer = 0)
+	. = ..()
+
+	var/obj/item/card/id/attached_id = parent
+	if(!attached_id)
+		return COMPONENT_INCOMPATIBLE
+
+	stored_trim = attached_id.trim
+	stored_assignment = attached_id.assignment
+
+
+	if(cooldown_timer)
+		on_cooldown = TRUE
+		addtimer(CALLBACK(src, PROC_REF(remove_cooldown)), cooldown_timer)
+
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attempt_unlock)
+
+/datum/component/off_duty_timer/Destroy(force, silent)
+	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
+	if(stored_trim)
+		qdel(stored_trim)
+		stored_trim = null
+
+	return ..()
+
+///Sets the on_cooldown variable to false, making it so that the ID can clock back in.
+/datum/component/off_duty_timer/proc/remove_cooldown()
+	on_cooldown = FALSE
+
+///Attempts an unlock if attacked by another ID. If the ID has HoP access, it will unlock and return TRUE
+/datum/component/off_duty_timer/proc/attempt_unlock(datum/source, obj/item/attacking_item, mob/user, params)
+	SIGNAL_HANDLER
+	if(!hop_locked)
+		return FALSE
+
+	var/obj/item/card/id/advanced/hop_id = attacking_item
+	if(!hop_id)
+		return FALSE
+
+	if(!(ACCESS_HOP in hop_id.access))
+		to_chat(user, span_warning("You lack the access to unlock [parent]"))
+		return FALSE
+
+	hop_locked = FALSE
+	to_chat(user, span_notice("[parent] has been unlocked, the owner is now able to clock in."))
+	log_admin("[parent] has been unlocked by [user] and is now able to be clocked back in.")
+
+	return TRUE
+
+/obj/item/wallframe/time_clock
+	name = "time clock frame"
+	desc = "Contains all of the parts needed to assemble a wall-mounted time clock"
+	icon_state = "unanchoredstatusdisplay"
+	result_path = /obj/machinery/time_clock
+	pixel_shift = 28
```

Comment:
Doesn't that just hard-shift it no matter the rotation? Won't that be an issue, unless it's not possible to rotate it, I guess.

Reply:
Code-wise, this works similar to intercoms when it comes to construction. So, I think it should be fine.

Reply:
![image](https://user-images.githubusercontent.com/68373373/233509999-a988eb1b-95ac-448a-81c3-131dfd24904e.png)

Reply:
Alright then, good to know.